app/lib/three_scale/policies/specification.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  module Policies
+    class Specification
+      extend ActiveModel::Naming
+      extend ActiveModel::Translation
+
+      def initialize(body)
+        @errors = ActiveModel::Errors.new(self)
+
+        begin
+          @doc = JSON.parse(body.to_s)
+        rescue JSON::ParserError
+          @errors.add(:base, :invalid_json)
+        end
+
+        @doc ||= {}
+      end
+
+      attr_reader :errors, :doc
+
+      JSON_SCHEMA = {'$ref' => 'http://apicast.io/policy-v1/schema#'}.freeze
+
+      def valid?
+        return false if errors.any?
+        JSONValidator.fully_validate(JSON_SCHEMA, doc).each { |error| errors.add(:base, error) }
+        errors.empty?
+      end
+
+      def self.setup_json_validator!
+        JSONValidator.setup!
+
+        schemas = Rails.root.join('app', 'lib', 'three_scale', 'policies', 'schemas', '*.schema.json')
```

Comment:
This can go to an initializer were you define which schemas are going to be registered.
One place and independent from the code.